### PR TITLE
[mlir][LLVM] Add exact flag

### PR DESCRIPTION
**Pull request by lfrenot**
The implementation is mostly based on the one existing for the nsw and nuw flags.

If the exact flag is present, the corresponding operation returns a poison value when the result is not exact. (For a division, if rounding happens; for a right shift, if a non-zero bit is shifted out.)

@zero9178, @gysit and @Dinistro, could you take a look?

### CODE REVIEW

File: mlir/lib/Target/LLVMIR/ModuleImport.cpp
Comment (Dinistro) [inline mlir/lib/Target/LLVMIR/ModuleImport.cpp:691]:
```suggestion
  iface.setIsExact(inst->isExact());
```

**Review by Dinistro — APPROVED** [review]
Thanks for this contribution 🙂 
Very nice work, LGTM % a small nit comment.

File: mlir/include/mlir/Dialect/LLVMIR/LLVMOps.td
Comment (gysit) [inline mlir/include/mlir/Dialect/LLVMIR/LLVMOps.td:79]:
```suggestion
class LLVM_IntArithmeticOpWithExactFlag<string mnemonic, string instName,
```
nit: Should we use ExactFlag to follow the naming of LLVM_IntArithmeticOpWithIsExact?

File: mlir/test/Dialect/LLVMIR/roundtrip.mlir
Comment (gysit) [inline mlir/test/Dialect/LLVMIR/roundtrip.mlir:52]:
```suggestion
// Integer exact flag.
```
ultra nit:

**Review by gysit — APPROVED** [review]
Thanks for adding the flag!

LGTM modulo nits.

File: mlir/include/mlir/Dialect/LLVMIR/LLVMOps.td
Comment (lfrenot) [reply mlir/include/mlir/Dialect/LLVMIR/LLVMOps.td]: Done!

File: mlir/test/Dialect/LLVMIR/roundtrip.mlir
Comment (lfrenot) [reply mlir/test/Dialect/LLVMIR/roundtrip.mlir]: Done!

**Comment by lfrenot** [comment]
If everything is fixed, is it possible for someone with write access to merge the PR?

**Comment by tobiasgrosser** [comment]
@Dinistro @gysit @zero9178, thank you for helping with this PR (and potentially merging it when ready).

**Pull request merged**
